app/models/usage_report.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2014 - 2023 UNICEF. All rights reserved.
+
+# class for Usage Report
+class UsageReport
+  include ActiveModel::Model
```

Comment:
Is there a reason that this is a Rails ActiveModel object rather than a plain Ruby object? Please move this business logic to a service, something like `UsageReportService`